experiments/Marmot/3dtools/3dtoolstest.tex

```diff
@@ -0,0 +1,50 @@
+\documentclass{article}
+\usepackage{tikz}
+\usepackage{tikz-3dplot} 
```

Comment:
Evidently this library is meant for interop with `tikz-3dplot`, which is why I won't merge this into mainline PGF/Ti*k*Z.  I'm not convinced by having more user macros.  I'm also not convinced by `\pgfmathtdparse`.  That seems very fragile.

Reply:
In response to https://github.com/pgf-tikz/pgf/issues/731#issuecomment-553616722 which was probably meant to appear here:

---

The 3dtools library introduces a whole bunch of new user macros, like `\pgfmathtdparse`, `\pgfmathprintvector`, `\tdparse` (is that the same as `\pgfmathtdparse`?) and `\orthmat` (is this meant for users? If not, why is it public?).  This somewhat collides with the PGF/Ti*k*Z philosophy of using keys and values instead of commands.

Right now, the whole 3dtools library is in a single file `tikzlibrary3dtools.code.tex`.  It uses the LaTeX package `calculator` and `\newcommand`, i.e. it cannot be used with other formats.  Also it defines macros with `\pgf...` prefix which clearly belong in a PGF rather than a Ti*k*Z library.

Last but not least, I am totally not stopping your from publishing this library independently on CTAN for use with Ti*k*Z, just like `tikz-cd` or `tikz-feynman`.  So while I can understand your disappointment that this is probably not going to be in mainline without some vast improvement, I am slightly confused by the accusation of “destroying hours of work”.

Reply:
I have packaged up `tikz-3dtools` and put it here: https://github.com/pgf-tikz/tikz-3dtools
Release v1.0 is also submitted to CTAN already.

Reply:
`tikz-3dtools` has appeared on CTAN and should be in TeX Live soon.
https://ctan.org/pkg/tikz-3dtools